frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -19,15 +19,84 @@ mod parse;
 
 use frame_support_procedural_tools::syn_ext as ext;
 use frame_support_procedural_tools::{generate_crate_access, generate_hidden_includes};
-use parse::{ModuleDeclaration, RuntimeDefinition, WhereSection};
+use parse::{ModuleDeclaration, RuntimeDefinition, WhereSection, ModulePart};
 use proc_macro::TokenStream;
-use proc_macro2::TokenStream as TokenStream2;
+use proc_macro2::{TokenStream as TokenStream2, Span};
 use quote::quote;
 use syn::{Ident, Result, TypePath};
+use std::collections::HashMap;
 
 /// The fixed name of the system module.
 const SYSTEM_MODULE_NAME: &str = "System";
 
+/// The complete definition of a module with the resulting fixed index.
+#[derive(Debug, Clone)]
+pub struct Module {
+	pub name: Ident,
+	pub index: u8,
+	pub module: Ident,
+	pub instance: Option<Ident>,
+	pub module_parts: Vec<ModulePart>,
+}
+
+impl Module {
+	/// Get resolved module parts
+	fn module_parts(&self) -> &[ModulePart] {
+		&self.module_parts
+	}
+
+	/// Find matching parts
+	fn find_part(&self, name: &str) -> Option<&ModulePart> {
+		self.module_parts.iter().find(|part| part.name() == name)
+	}
+
+	/// Return whether module contains part
+	fn exists_part(&self, name: &str) -> bool {
+		self.find_part(name).is_some()
+	}
+}
+
+/// Convert from the parsed module to their final information.
+/// Assign index to each modules using same rules as rust for fieldless enum.
+/// I.e. implicit are assigned number incrementedly from last explicit or 0.
+fn complete_modules(decl: impl Iterator<Item = ModuleDeclaration>) -> syn::Result<Vec<Module>> {
+	let mut indices = HashMap::new();
+	let mut last_index: Option<u8> = None;
+
+	decl
+		.map(|module| {
+			let final_index = match module.index {
+				Some(i) => i,
+				None => last_index.map_or(Some(0), |i| i.checked_add(1))
+					.ok_or_else(|| {
+						let msg = "Module index doesn't fit into u8, index is 256";
+						syn::Error::new(module.name.span(), msg)
+					})?,
+			};
+
+			last_index = Some(final_index);
+
+			if let Some(used_module) = indices.insert(final_index, module.name.clone()) {
+				let msg = format!(
+					"Module index are conflicting both modules {} and {} are at index {}",
+					module.name,
+					used_module,
+					final_index,
+				);
+				return Err(syn::Error::new(module.name.span(), msg));
```

Comment:
https://docs.rs/syn/1.0.41/syn/struct.Error.html#method.combine please use this here to print errors at both places.

Reply:
done here https://github.com/paritytech/substrate/pull/6969/commits/95ef421253368d06a16004ca744bdd0179a4d474